### ISSUE TITLE
Train data preprocess

### ISSUE THREAD

**Issue by kunibald413** [index-tts/index-tts#229]
I am currently looking into **continue training the `UnifiedVoice` model**. Its `forward` method requires `mel_codes` as input:

```python
def forward(self, speech_conditioning_latent, text_inputs, text_lengths, mel_codes, ...)
```

My understanding is that these `mel_codes` are generated by a pre-trained `DiscreteVAE` from mel-spectrograms. I try to understand the correct preprocessing steps for audio that lead to these inputs.

I've found the `dvae_wav_to_mel` function that I assume creates the input for the `DiscreteVAE`:

```python
def dvae_wav_to_mel(
    wav, mel_norms_file="../experiments/clips_mel_norms.pth", mel_norms=None, device=torch.device("cpu")
):
    mel_stft = torchaudio.transforms.MelSpectrogram(
        n_fft=1024,
        hop_length=256,
        win_length=1024,
        power=2,
        normalized=False,
        sample_rate=22050,
        f_min=0,
        f_max=8000,
        n_mels=80,
        norm="slaney",
    ).to(device)
    wav = wav.to(device)
    mel = mel_stft(wav)
    mel = torch.log(torch.clamp(mel, min=1e-5))
    if mel_norms is None:
        mel_norms = torch.load(mel_norms_file, map_location=device)
    mel = mel / mel_norms.unsqueeze(0).unsqueeze(-1)
    return mel
```

Currenlty this it is unclear to me as to how to prepare training data correctly for `UnifiedVoice`:

1.  **Missing `mel_norms.pth` File:** The `dvae_wav_to_mel` function relies on a `mel_norms.pth` file for a crucial normalization step. This file appears to be **missing** from the public repository. Without it, I cannot replicate the exact data preprocessing that was used to generate the `mel_codes` that `UnifiedVoice` was originally trained on.

2.  **Inconsistent Mel-Spectrogram Parameters:** I've noticed a discrepancy in mel-spectrogram parameters between the `dvae_wav_to_mel` function and the `dataset.mel` configuration provided in your `config.yaml` (which I believe describes the actual training dataset's preprocessing):

    **`dataset.mel` config from `config.yaml`:**

    ```yaml
    dataset:
        # ... other params
        mel:
            sample_rate: 24000
            n_fft: 1024
            hop_length: 256
            win_length: 1024
            n_mels: 100
            mel_fmin: 0
            normalize: false
    ```

    The `dvae_wav_to_mel` function uses `sample_rate=22050` and `n_mels=80`, while the `dataset.mel` config (from the training setup) specifies `sample_rate=24000` and `n_mels=100`. Ensuring that the `mel_codes` are generated using the **exact same preprocessing parameters** as the original training data is paramount for `UnifiedVoice` to learn effectively.

Could you please clarify how to prepare the data:

1.  **Provide the `mel_norms.pth` file** (or detailed instructions on how to generate it accurately) that corresponds to the training dataset. This file is essential for replicating the exact normalization used during the original training data preparation.
2.  **Clarify the definitive, exact mel-spectrogram preprocessing parameters (including all `torchaudio.transforms.MelSpectrogram` arguments like `power`, `norm`, `f_max`, and post-processing steps)** that were used to create the `mel_codes` for the `UnifiedVoice` training dataset.

Thank you very much for your time!

**Comment by kunibald413** [comment]
`dvae_wav_to_mel` seems to be fine to use with right params
and skipping normalization seems to work as well

**Comment by yrom** [comment]
You can take a look the `process_audio_data` function for preprocessing Mel-codes in this notebook [preprocess_mel_dataset.ipynb](https://github.com/yrom/finetune-index-tts/blob/main/preprocess_mel_dataset.ipynb)
for reference only, may contain mistakes

**Comment by yurtmete** [comment]
> `dvae_wav_to_mel` seems to be fine to use with right params and skipping normalization seems to work as well

@kunibald413 may I ask what do you mean by right params here? Do you mean the parameters in **dataset.mel** in **config.yaml** or the ones described in **`indextts/vqvae/xtts_dvae.py`** as in the following:

`mel_stft = torchaudio.transforms.MelSpectrogram(
        n_fft=1024,
        hop_length=256,
        win_length=1024,
        power=2,
        normalized=False,
        sample_rate=22050,
        f_min=0,
        f_max=8000,
        n_mels=80,
        norm="slaney",
    ).to(device)`

**Comment by kunibald413** [comment]
> > `dvae_wav_to_mel` seems to be fine to use with right params and skipping normalization seems to work as well
> 
> [@kunibald413](https://github.com/kunibald413) may I ask what do you mean by right params here? Do you mean the parameters in **dataset.mel** in **config.yaml** or the ones described in **`indextts/vqvae/xtts_dvae.py`** as in the following:
> 
> `mel_stft = torchaudio.transforms.MelSpectrogram( n_fft=1024, hop_length=256, win_length=1024, power=2, normalized=False, sample_rate=22050, f_min=0, f_max=8000, n_mels=80, norm="slaney", ).to(device)`

just the default ones, but this is already solved. a few traing repos already out
